lnwallet/reservation.go

```diff
@@ -416,6 +426,10 @@ func NewChannelReservation(capacity, localFundingAmt btcutil.Amount,
 		chanType |= channeldb.ScidAliasFeatureBit
 	}
 
+	if req.CommitType == CommitmentTypeSimpleTaprootOverlay {
+		chanType |= channeldb.TapscriptRootBit
+	}
+
```

Comment:
So that part was added before we had the rest of the scaffolding, just to test the `musig2` layer. I think we can add a sanity check here to make sure the overlay bit isn't set in isolation as you mention.